packages/contracts/contracts/marketplace/v00/RestrictableContract.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity ^0.4.24;
+
+import "../../../node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+// import "../../../node_modules/openzeppelin-solidity/contracts/AddressUtils.sol";
+
+/**
+ * @title RestrictableContract
+ * @author Shahul Hameed <hello@hameid.net>
+ *
+ * Extends Ownable contract and maintains a whitelist of users and whitelist of contracts
+ */
+
+contract RestrictableContract is Ownable {
+    mapping(address => bool) public allowedContracts;
+    mapping(address => bool) public allowedUsers;
```

Comment:
What type of use cases related to a P2P marketplace do you anticipate would need a whitelist of users ?

Reply:
I added that just in case if whitelisting of sellers/buyers are needed. Can remove it if it is not necessary.